modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/model/listener/LayoutModelListener.java

```diff
@@ -67,14 +79,60 @@ public void onAfterRemove(Layout layout) {
 					layoutLocalization);
 			}
 		}
+
+		clearCache(layout);
+	}
+
+	@Override
+	public void onAfterUpdate(Layout originalLayout, Layout layout) {
+		clearCache(layout);
 	}
 
 	@Override
 	public void onBeforeRemove(Layout layout) throws ModelListenerException {
 		_layoutClassedModelUsageLocalService.
 			deleteLayoutClassedModelUsagesByPlid(layout.getPlid());
+
+		try {
+			if ((layout == null) ||
```

Comment:
@julschong This null check, if being put here, is redundant, because the method call above uses `layout.getPlid()` which will throw NPE when layout is null.

All other methods are performing the null check at the beginning, so I think we should either extract this null check to the beginning of the method, or put this moved logic before the `_layoutClassedModelUsageLocalService` call.